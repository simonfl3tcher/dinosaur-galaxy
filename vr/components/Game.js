// @flow

// React
import React, { Component } from 'react';
import {
  Text,
  View
} from 'react-vr';

// Components
import Question from './Question';

// Libs
import PropTypes from 'prop-types';
import styles from '../styles/main';

export default class Game extends Component {
  render() {
    return (
      <View style={styles.gameStyle}>
        <Text style={styles.text}>VRSarus!</Text>
        <Text style={styles.text}>Current Score: {this.props.score}</Text>
        <Text style={styles.text}>
          Highest Score: {this.props.highestScore}
        </Text>
        <Question
          question={this.props.question}
          pickAnswer={this.props.pickAnswer} />
      </View>
    );
  }
}

Game.propTypes = {
  highestScore: PropTypes.number,
  pickAnswer: PropTypes.func,
  question: PropTypes.object,
  score: PropTypes.number
};
