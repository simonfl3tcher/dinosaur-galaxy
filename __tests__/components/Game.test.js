import 'react-native';
import 'react-vr';
import React from 'react';
import Game from '../../vr/components/Game.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const question = {
    question: 'example?',
    answers: {
      a: 'a',
      b: 'b',
      c: 'b',
    },
    answer: 'a',
  };

  const tree = renderer
    .create(<Game score={0} highestScore={0} question={question} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
