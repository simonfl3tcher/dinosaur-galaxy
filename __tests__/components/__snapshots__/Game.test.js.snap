// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "transform": Array [
        Object {
          "translate": Array [
            -2.25,
            0,
            0,
          ],
        },
      ],
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    isOnLayer={false}
    style={
      Object {
        "color": "#fff",
        "fontSize": 0.2,
        "textAlign": "center",
        "transform": Array [
          Object {
            "translate": Array [
              0,
              2,
              -5,
            ],
          },
        ],
      }
    }
  >
    VRSarus!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    isOnLayer={false}
    style={
      Object {
        "color": "#fff",
        "fontSize": 0.2,
        "textAlign": "center",
        "transform": Array [
          Object {
            "translate": Array [
              0,
              2,
              -5,
            ],
          },
        ],
      }
    }
  >
    Current Score: 
    0
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    isOnLayer={false}
    style={
      Object {
        "color": "#fff",
        "fontSize": 0.2,
        "textAlign": "center",
        "transform": Array [
          Object {
            "translate": Array [
              0,
              2,
              -5,
            ],
          },
        ],
      }
    }
  >
    Highest Score: 
    0
  </Text>
  <View>
    <View
      disabled={false}
      ignoreLongClick={false}
      onClick={[Function]}
      onEnter={[Function]}
      onExit={[Function]}
      onInput={[Function]}
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        isOnLayer={false}
        style={
          Object {
            "backgroundColor": "#fff",
            "color": "#000",
            "fontSize": 0.2,
            "margin": 0.1,
            "textAlign": "center",
            "transform": Array [
              Object {
                "translate": Array [
                  0,
                  2,
                  -5,
                ],
              },
            ],
          }
        }
      >
        a
      </Text>
    </View>
    <View
      disabled={false}
      ignoreLongClick={false}
      onClick={[Function]}
      onEnter={[Function]}
      onExit={[Function]}
      onInput={[Function]}
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        isOnLayer={false}
        style={
          Object {
            "backgroundColor": "#fff",
            "color": "#000",
            "fontSize": 0.2,
            "margin": 0.1,
            "textAlign": "center",
            "transform": Array [
              Object {
                "translate": Array [
                  0,
                  2,
                  -5,
                ],
              },
            ],
          }
        }
      >
        b
      </Text>
    </View>
    <View
      disabled={false}
      ignoreLongClick={false}
      onClick={[Function]}
      onEnter={[Function]}
      onExit={[Function]}
      onInput={[Function]}
      testID={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        isOnLayer={false}
        style={
          Object {
            "backgroundColor": "#fff",
            "color": "#000",
            "fontSize": 0.2,
            "margin": 0.1,
            "textAlign": "center",
            "transform": Array [
              Object {
                "translate": Array [
                  0,
                  2,
                  -5,
                ],
              },
            ],
          }
        }
      >
        b
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      isOnLayer={false}
      style={
        Object {
          "backgroundColor": "#fff",
          "color": "#000",
          "fontSize": 0.5,
          "textAlign": "center",
          "transform": Array [
            Object {
              "translate": Array [
                0,
                1,
                -5,
              ],
            },
          ],
        }
      }
    >
      example?
    </Text>
  </View>
</View>
`;
